interview-questions.md

```diff
@@ -5,40 +5,64 @@ Answer the following questions. First, without external resources. Challenge you
 1. Why would you use the method super()?
 
   Your answer:
+      Super is used in conjunction with "constructor" which will call functions from a parent class.
 
   Researched answer:
+      When used in a constructor, the super keyword appears alone and must be used before the "this" keyword is used. The super keyword can also be used to call functions on a parent object. super() will call the constructor of its parent class. This is required when you need to access some variables from the parent class. In React, when you call super with props. React will make props available across the component through this.props .
+        - taken from LearnAcademy lesson plan + MDN
 
 
 
 2. What is a virtual DOM?
-
   Your answer:
+      I think it stands for domain object manager and its purpose is to manage how the code translates to the interface.
 
   Researched answer:
+      In React, for every DOM object, there is a corresponding “virtual DOM object.” A virtual DOM object is a representation of a DOM object, like a lightweight copy.
+
+      A virtual DOM object has the same properties as a real DOM object, but it lacks the real thing’s power to directly change what’s on the screen.
+
+      Manipulating the DOM is slow. Manipulating the virtual DOM is much faster, because nothing gets drawn onscreen. Think of manipulating the virtual DOM as editing a blueprint, as opposed to moving rooms in an actual house.
+        - taken from CodeCademy!
 
 
 
 3. When creating a basic (stateless) class component in React, what are the necessary elements needed to display "Hello World" in the browser?
 
   Your answer:
+        class Name extends Component {
+          render (){
+            <h1> Hello World</h1>
+          }
+        }
 
   Researched answer:
-
-
+        class Name extends React.Component {
+          render (){
+            return(
+              <h1> Hello World</h1>
+              )
+            }
+          }
 
 4. What is JSX?
 
   Your answer:
+      It's a special type of javascript language used in React.
 
   Researched answer:
+      JSX stands for JavaScript XML. JSX allows us to write HTML in React. JSX makes it easier to write and add HTML in React.
+        - taken from W3Schools
 
 
 
 5. What is state in React?
 
   Your answer:
+      I still don't really understand what state is.
```

Comment:
And that's totally okay and I appreciate the honesty! 

In an interview setting like this, if you were asked this question I think it would be important to follow up with something like: "I'm not sure about state, but I would take a look into the react documentation to find out!" as this would show the person you are interviewing with that you are proactive in finding the answer and are resourceful in your ability to learn. 

Bryan will totally go over a lot of this in PD week and when you're getting more interview practice later on, but I just want to give you a leg up on everyone else and provide some cheat codes to these interview questions if you are ever in this situation ;D